src/tunnel.cc

```diff
@@ -491,6 +491,9 @@ TunnelStateData::retryOrBail(const char *context)
         return sendError(savedError, bailDescription ? bailDescription : context);
     *status_ptr = savedError->httpStatus;
 
+    if (request)
+        request->hier.stopPeerClock(false);
```

Comment:
Why are we moving this call? On the surface, the move adds sendError() delays to `%<tt` but sending error to the client is not a Squid-peer activity. The Squid-to-peer connection is gone here, in retryOrBail(), so there can be no Squid-peer communication going on. It feels like the original stopPeerClock() call was in the best possible place.

If we are moving it just so that we do not call stopPeerClock() twice in sendError() case, I would not move; it only complicates logic/dependencies. It is better to call stopPeerClock() an extra time then to miss that call in sendError() refactoring. This is especially true since "sending error", in general, does not imply that we are done with peer communication -- the reader may not expect sendError() to stopPeerClock(). The existing tunneling code does not do that, but we could be, in theory, sending an error to the client while still writing to the server...

N.B. sendError() should continue to call stopPeerClock(). A sendError() call currently represents the last TunnelStateData activity in all callers AFAICT.